### PR TITLE
[Merged by Bors] - feat: Continuous real-valued functions separate points in T3.5-spaces.

### PR DESCRIPTION
**Pull request by kkytola**
Continuous real-valued functions separate points in T3.5-spaces, spelled in terms of `SeparatesPoints`. There was a brief [Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/separatesPoints_continuous/near/430510853).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/CompletelyRegular.lean [hunk 1]
```diff
@@ -95,3 +95,10 @@ instance T4Space.instT35Space [T4Space X] : T35Space X := by
   have : T1Space X := T2Space.t1Space
   have : CompletelyRegularSpace X := NormalSpace.instCompletelyRegularSpace
   exact {}
+
+lemma separatesPoints_continuous_of_T35Space [T35Space X] :
+    SeparatesPoints (Continuous : Set (X → ℝ)) := by
+  intro x y x_ne_y
+  obtain ⟨f, f_cont, f_zero, f_one⟩ :=
+    CompletelyRegularSpace.completely_regular x {y} isClosed_singleton x_ne_y
+  refine ⟨fun x ↦ f x, continuous_subtype_val.comp f_cont, by aesop⟩
```
Comment (mattdiamond) [inline Mathlib/Topology/CompletelyRegular.lean:104]: Should this be spelled `exact` instead of `refine` or is there a benefit to using `refine` here?

Comment (Parcly-Taxel) [reply Mathlib/Topology/CompletelyRegular.lean:104]: `exact` can be used here, and is preferred over `refine` to _close_ a proof.
```suggestion
  exact ⟨fun x ↦ f x, continuous_subtype_val.comp f_cont, by aesop⟩
```

Comment (kkytola) [reply Mathlib/Topology/CompletelyRegular.lean]: Thanks! (And sorry about that sloppiness in such a small PR.)

**Comment by mapehe** [comment]
Looks neat to me 💎

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**